// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

syntax = "proto3";

package flwr.proto;
import "flwr/proto/run.proto";
import "flwr/proto/node.proto";

message Account {
  string id = 1;
  string name = 2;
}

message FederationAccount {
  Account account = 1;
  string role = 2;
}

message Federation {
  string name = 1;
  repeated string member_aids = 2; // Deprecated in v1.26.0
  repeated NodeInfo nodes = 3;
  repeated Run runs = 4;
  string description = 5;                  // Added in v1.26.0
  repeated FederationAccount accounts = 6; // Added in v1.26.0
}
